libfreerdp/core/window.c

```diff
@@ -295,9 +296,14 @@ BOOL update_read_window_state_order(wStream* s, WINDOW_ORDER_INFO* orderInfo, WI
 		Stream_Read_UINT16(s, windowState->numWindowRects); /* numWindowRects (2 bytes) */
 
 		size = sizeof(RECTANGLE_16) * windowState->numWindowRects;
-		windowState->windowRects = (RECTANGLE_16*) malloc(size);
-		if (!windowState->windowRects)
+		newRect = (RECTANGLE_16*)realloc(windowState->windowRects, size);
+		if (!newRect)
```

Comment:
This can cause a double free if windowState->numWindowRects is 0. (And also breaks rails).